syne_tune/blackbox_repository/conversion_scripts/scripts/yahpo_import.py

```diff
@@ -0,0 +1,271 @@
+        else:
+            """
+            copying the parent comment of the parent class:
+            "not passing a fidelity is possible if either the blackbox does not have a fidelity space
+            or if it has a single fidelity in its fidelity space. In the latter case, all fidelities are returned in form
+            of a tensor with shape (num_fidelities, num_objectives)."
+            This is used for efficiency (it is much faster to retrieve a full row in an array in term of read time).
+            """
+            # returns a tensor of shape (num_fidelities, num_objectives)
+            num_fidelities = len(self.fidelity_values)
+            num_objectives = len(self.objectives_names)
+            result = np.empty((num_fidelities, num_objectives))
+            fidelity_name = next(iter(self.fidelity_space.keys()))
+            configs = []
+            for fidelity in self.fidelity_values:
+                config_with_fidelity = configuration.copy()
+                config_with_fidelity[fidelity_name] = fidelity
+                configs.append(config_with_fidelity)
+            result_dicts = self.benchmark.objective_function(configs, seed=seed)
+
+            for i, fidelity in enumerate(self.fidelity_values):
+                result[i] = [
+                    result_dicts[i][objective] for objective in self.objectives_names
+                ]
+
+            return result
+
+    def set_instance(self, instance):
+        """
+        Set an instance for the underlying YAHPO Benchmark.
+        """
+        # Set the instance in the benchmark
+        self.benchmark.set_instance(instance)
+        # Update the configspace with the fixed instance
+        if self.benchmark.config.instance_names:
+            instance_names = self.benchmark.config.instance_names
+        else:
+            instance_names = "instance-names"
+        self.configuration_space[instance_names] = cs.choice([instance])
+        return self
+
+    @property
+    def instances(self) -> np.array:
+        return self.benchmark.instances
+
+    @property
+    def fidelity_values(self) -> np.array:
+        fids = next(iter(self.fidelity_space.values()))
+        return np.arange(fids.lower, fids.upper)
+
+    @property
+    def time_attribute(self) -> str:
+        """Name of the time column"""
+        return self.benchmark.config.runtime_name
+
+
+def cs_to_synetune(config_space):
+    """
+    Convert ConfigSpace.ConfigSpace to a synetune configspace.
+
+    This should probably either be improved or we could also dump the 'synetune-configspaces' in 'yahpo_data'
+    like we did for R (paradox) configspaces.
+    Currently this does not cover all possible hyperparameters I think.
+    """
+    hps = config_space.get_hyperparameters()
+
+    keys = []
+    vals = []
+    for a in hps:
+        keys += [a.name]
+        if isinstance(a, ConfigSpace.hyperparameters.CategoricalHyperparameter):
```

Comment:
Could we throw an exception if there is a hyperparameter type that is not covered?